src/yunohost/permission.py

```diff
@@ -0,0 +1,510 @@
+
+logger = getActionLogger('yunohost.user')
+
+def user_permission_list(auth, app=None, permission=None, username=None, group=None):
+    """
+    List permission for specific application
+
+    Keyword argument:
+        app        -- an application OR sftp, xmpp (metronome), mail
+        permission -- name of the permission ("main" by default)
+        username   -- Username to get informations
+        group      -- Groupname to get informations
+
+    """
+
+    user_l = user_list(auth, ['uid'])['users']
+
+    permission_attrs = [
+        'cn',
+        'groupPermission',
+        'inheritPermission',
+        'URL',
+    ]
+
+    # Normally app is alway defined but it should be possible to set it
+    if app and not isinstance(app, list):
+        app = [app]
+    if permission and not isinstance(permission, list):
+        permission = [permission]
+    if not isinstance(username, list):
+        username = [username]
+    if not isinstance(group, list):
+        group = [group]
+
+    permissions = {}
+
+    result = auth.search('ou=permission,dc=yunohost,dc=org',
+                         '(objectclass=permissionYnh)', permission_attrs)
+
+    for res in result:
+        permission_name = res['cn'][0].split('.')[0]
+        try:
+            app_name = res['cn'][0].split('.')[1]
+        except:
+            logger.warning(m18n.n('permission_name_not_valid', permission=per))
+        group_name = []
+        if 'groupPermission' in res:
+            for g in res['groupPermission']:
+                group_name.append(g.split("=")[1].split(",")[0])
+        user_name = []
+        if 'inheritPermission' in res:
+            for u in res['inheritPermission']:
+                user_name.append(u.split("=")[1].split(",")[0])
+
+        # Don't show the result if the user diffined a specific permission, user or group
+        if app and not app_name in app:
+            continue
+        if permission and not permission_name in permission:
+            continue
+        if username[0] and not set(username) & set(user_name):
+            continue
+        if group[0] and not set(group) & set(group_name):
+            continue
+
+        if not app_name in permissions:
+            permissions[app_name] = {}
+
+        permissions[app_name][permission_name] = {'allowed_users':[], 'allowed_groups':[]}
+        for g in group_name:
+            permissions[app_name][permission_name]['allowed_groups'].append(g)
+        for u in user_name:
+            permissions[app_name][permission_name]['allowed_users'].append(u)
+        if 'URL' in res:
+            permissions[app_name][permission_name]['URL'] = []
+            for u in res['URL']:
+                permissions[app_name][permission_name]['URL'].append(u)
+
+    return {'permissions': permissions}
+
+
+def user_permission_update(operation_logger, auth, app=[], permission=None, add_username=None, add_group=None, del_username=None, del_group=None, sync_perm=True):
```

Comment:
Same question : shouldn't this be available from the actionmap ? Or maybe it is available through the `yunohost app [addaccess|removeaccess|...]` but it's a bit confusing ...

Reply:
Well, this function is a merge of `yunohost user permission add, remove and clear`. So it should be not needed.

By example it's used here : https://github.com/YunoHost/yunohost/pull/585/files#diff-f4b72011f32f548ac5655aec63be4d7fR734